lib/HLSL/HLSignatureLower.cpp

```diff
@@ -1715,6 +1759,12 @@ void HLSignatureLower::GenerateGetMeshPayloadOperation() {
 }
 // Lower signatures.
 void HLSignatureLower::Run() {
+
+  // Generate error and exit if semantic type
+  // is not one of the allowed types
+  if (!ValidateSemanticType(Entry))
+    return;
```

Comment:
This provides no indication of failure to the subsequent passes, I worry this will bail out of the rest of this pass and potentially cause weird followup errors on subsequent passes that expect these operations to have been performed.

Reply:
That's true, and I have a feeling that continuing with this pass will result in fewer weird cascading errors from this than returning here.  Currently, it seems the only way to abort the pipeline is with a report_fatal_error, which currently raises a structured exception - which is way too harsh for something like this, and we should likely change report_fatal_error to throw a C++ exception, since we still have a chance of emitting a useful error message then.

Reply:
> I have a feeling that continuing with this pass will result in fewer weird cascading errors from this than returning here

Actually bailing out early avoids crashes at least in some cases. Like for CS SV semantics, if we pass an invalid type like `float` and let it continue with the lowering then it crashes in  `HLSignatureLower::GenerateDxilCSInputs()` in the below line.

```
    // If the argument is of non-i32 type, convert here
    if (newArg->getType() != NumTy)
      newArg = Builder.CreateZExtOrTrunc(newArg, NumTy);
```

Based on the comments here, I agree we need a mechanism to stop further compilation when we generate errors. I suspect we would need that mechanism not just in this change, but probably at other places too where we generate errors. Therefore I created an issue #3496 to track this as a future enhancement that we would like to implement.

Reply:
That's consistent with my later investigations. I approve of solving the larger problem right later.